src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,122 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useContext, useState } from 'react';
+import { createComment } from '../api/comments';
+import { SelectedPostContext } from '../providers/PostProvider';
+import { Comment } from '../types/Comment';
+
+const EMAIL_REGEX = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+
+type Props = {
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ setComments }) => {
+  const [authName, setAuthName] = useState('');
+  const [authNameError, setAuthNameError] = useState('');
+  const [authEmail, setAuthEmail] = useState('');
+  const [authEmailError, setAuthEmailError] = useState('');
+  const [authComment, setAuthComment] = useState('');
+  const [authCommentError, setAuthCommentError] = useState('');
```

Comment:
Better combine it into 2 states
const [author, setAuthor] = useState({}); // { name: "Name",  email: "my@email", comment: "test comment" }
const [error, setError] = useState({}); // { name: "",  email: "", comment: "" }